### PR TITLE
adding cmdlets for edge storage account and edge storage container

### PR DESCRIPTION
**Pull request by raviten**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

[Design Issue](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/443)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by raviten** [comment]
[Design Issue](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/443)

**Comment by raviten** [comment]
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/443

File: src/DataBoxEdge/DataBoxEdge/Common/DataBoxEdgeResourceIdentifier.cs [hunk 1]
```diff
@@ -23,10 +24,10 @@ public class DataBoxEdgeResourceIdentifier : ResourceIdentifier
         public bool IsSubResource { get; }
         public string DeviceName { get; }
         public string Name { get; }
-
+
```
Comment (msJinLei) [inline src/DataBoxEdge/DataBoxEdge/Common/DataBoxEdgeResourceIdentifier.cs]: Unnecessary changed line.

File: src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeNetworkAdapter.cs [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+using System;
+using NetworkAdapter = Microsoft.Azure.Management.DataBoxEdge.Models.NetworkAdapter;
 
 namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models
 {
     public class PSDataBoxEdgeNetworkAdapter
     {
-        [Ps1Xml(Label = "Name", Target = ViewControl.Table,
-            ScriptBlock = "$_.networkAdapter.NetworkAdapterName", Position = 1)]
+
```
Comment (msJinLei) [inline src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeNetworkAdapter.cs]: Empty new line

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageAccounts/DataBoxEdgeStorageAccountNewCmdlet.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        public string DeviceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = Constants.NameHelpMessage,
+            Position = 2)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = EdgeStorageAccountParameterSet,
```
[the remark this answers is not in the document]
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageAccounts/DataBoxEdgeStorageAccountNewCmdlet.cs]: Yep, this is not required, will remove it

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageContainers/DataBoxEdgeStorageContainerNewCmdlet.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                throw new Exception(GetResourceAlreadyExistMessage());
+            }
+            catch (CloudException e)
+            {
+                if (e.Response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return false;
+                }
+
+                throw;
```
[the remark this answers is not in the document]
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageContainers/DataBoxEdgeStorageContainerNewCmdlet.cs]: Here, only not found exception needs to be captured while in any other case throw the exception. In general throw keyword throws the last exception if not handled

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageAccounts/DataBoxEdgeStorageAccountNewCmdlet.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                throw new Exception(GetResourceAlreadyExistMessage());
+            }
+            catch (CloudException e)
+            {
+                if (e.Response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return false;
+                }
+
+                throw;
```
[the remark this answers is not in the document]
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageAccounts/DataBoxEdgeStorageAccountNewCmdlet.cs]: Here, only not found exception needs to be captured while in any other case throw the previous exception. In general throw keyword throws the last exception if not handled, calling throw will throw high level exception which is CloudException only

[the remark this answers is not in the document]
Comment (msJinLei) [reply src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageAccounts/DataBoxEdgeStorageAccountNewCmdlet.cs]: > Here, only not found exception needs to be captured while in any other case throw the previous exception. In general throw keyword throws the last exception if not handled, calling throw will throw high level exception which is CloudException only

Cool. Thanks for explanation.

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageContainers/DataBoxEdgeStorageContainerNewCmdlet.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (msJinLei) [reply src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/EdgeStorageContainers/DataBoxEdgeStorageContainerNewCmdlet.cs]: > Here, only not found exception needs to be captured while in any other case throw the exception. In general throw keyword throws the last exception if not handled

Cool. Thanks for explanation.

**Pull request merged**